plugins/lsp/language-servers/default.nix

```
@@ -238,6 +238,12 @@ let
       serverName = "java_language_server";
       cmd = cfg: [ "${cfg.package}/bin/java-language-server" ];
     }
+    {
+      name = "jdt-language-server";
+      description = "Eclipse JDT Language Server for Java";
+      serverName = "jdtls";
+      cmd = cfg: [ "${cfg.package}/bin/jdtls" ];
```

Any reason this can't just be:
```suggestion
      cmd = cfg: [ (lib.getExe cfg.package) ];
```
?

I'd assume is the default when `cmd` is not specified, so is this line even needed?

If it isn't the default, maybe it should be :thinking:

---

I do not see `lib.getExe` used anywhere in the existing list of language servers, and looked through `_mk-lsp.nix` and `lsp/default.nix` and it doesn't seem to perform that mapping as a default. If you feel strongly I can take a stab at a new revision.

---

Looks like the default is `cfg: null`, so currently it is needed.

It gets used here: https://github.com/nix-community/nixvim/blob/216d64c158da5523d5b3db0895e1345175c21502/plugins/lsp/language-servers/_mk-lsp.nix#L57

I think it can be improved but not in this PR.

I don't see why `lib.getExe` can't be used though...

I suspect it initially wasn't known about and the precedent of not using it has just been followed without question so far.

---

Yeah I think it's exactly the reason. lib.getExe is more maintainable than hardcoding the executable name

---

I added this to the change when I rebased.